pkg/proxy/apis/kubeproxyconfig/validation/validation.go

```diff
@@ -141,15 +142,38 @@ func validateKubeProxyConntrackConfiguration(config kubeproxyconfig.KubeProxyCon
 }
 
 func validateProxyMode(mode kubeproxyconfig.ProxyMode, fldPath *field.Path) field.ErrorList {
+	if runtime.GOOS == "windows" {
+		return validateProxyModeWindows(mode, fldPath)
+	}
+
+	return validateProxyModeLinux(mode, fldPath)
```

Comment:
Instead of checking GOOS, could we perhaps have separate validation_windows.go and validation_linux.go files with +build directives instead?  That seems to be a normal pattern, at least more normal than checking GOOS.